mux.go

```diff
@@ -45,27 +45,27 @@ func Group(s *Server, path string, routeOptions ...func(*BaseRoute)) *Server {
 }
 
 // Capture all methods (GET, POST, PUT, PATCH, DELETE) and register a controller.
-func All[ReturnType, Body any, Contexted ctx[Body]](s *Server, path string, controller func(Contexted) (ReturnType, error), options ...func(*BaseRoute)) *Route[ReturnType, Body] {
+func All[ReturnType, Body any, Contexted ctxWithBody[Body]](s *Server, path string, controller func(Contexted) (ReturnType, error), options ...func(*BaseRoute)) *Route[ReturnType, Body] {
```

Comment:
There is something strange here

All is exported, but ctxWithBody is not.

It won't be easy for package users to pass a object from this interface.

They won't be able to build their own context, but they won't be able to check their implementation satisfies the expected interface

I'm talking about doing:

var _ ctxWithBody = Whatever{}

Reply:
But it was already like this 🤔

Reply:
Yes they're not meant to check their implementation because they must not implement it, we're providing the contexts that are used by users.

I must refactor Contexts but this is a breaking change...